dot/state/base.go

```diff
@@ -154,3 +154,26 @@ func (s *BaseState) loadFirstSlot() (uint64, error) {
 
 	return binary.LittleEndian.Uint64(data), nil
 }
+
+// storePruningData stores the pruner configuration.
+func (s *BaseState) storePruningData(mode pruner.Config) error {
+	encMode, err := json.Marshal(mode)
+	if err != nil {
+		return fmt.Errorf("cannot scale encode pruning mode: %s", err)
+	}
+
+	return s.db.Put(common.PruningKey, encMode)
+}
+
+// loadPruningData retrieves pruner configuration from db.
+func (s *BaseState) loadPruningData() (pruner.Config, error) {
+	data, err := s.db.Get(common.PruningKey)
```

Comment:
@arijitAD I hope you are still amongst us 👻 !

I am reworking the pruning code (it needs to be modified to support the newer v1 state trie & prune subvalues).
I have been through all the conversations on this PR but couldn't find an answer; why do we load and store the pruning config in the base state? Do you see any problem with just relying on the user `--pruning` flag (or toml config)?